packages/hierarchy-builder/src/test/internal/operators/grouping/PropertiesGrouping.test.ts

```diff
@@ -0,0 +1,1482 @@
+        };
+        expect(await propertiesGrouping.createPropertyGroups(metadataProvider, nodes, propertyInfo, formatter)).to.deep.eq({
+          groupingType: "property",
+          grouped: [],
+          ungrouped: nodes,
+        });
+      });
+    });
+
+    it("doesn't group when byProperties isn't set", async () => {
+      const nodes = [
+        createTestProcessedInstanceNode({
+          key: { type: "instances", instanceKeys: [{ className: "TestSchema.Class", id: "0x1" }] },
+        }),
+      ];
+      const ecClass = { fullName: "TestSchema.Class" } as unknown as ECClass;
+      const propertyInfo: propertiesGrouping.PropertyGroupInfo = {
+        ecClass,
+        previousPropertiesGroupingInfo: [],
+        propertyGroup: { propertyName: "PropertyName" },
+      };
+      expect(await propertiesGrouping.createPropertyGroups(metadataProvider, nodes, propertyInfo, formatter)).to.deep.eq({
+        groupingType: "property",
+        grouped: [],
+        ungrouped: nodes,
+      });
+    });
+
+    describe("Range isn't provided", async () => {
+      it("doesn't group node, when property value is not primitive", async () => {
+        const nodes = [
+          createTestProcessedInstanceNode({
+            key: { type: "instances", instanceKeys: [{ className: "TestSchema.Class", id: "0x1" }] },
+            processingParams: {
+              grouping: {
+                byProperties: {
+                  fullClassName: "TestSchema.Class",
+                  propertyGroups: [{ propertyName: "PropertyName", propertyValue: "PropertyValue" }],
+                },
+              },
+            },
+          }),
+        ];
+        const ecClass = { fullName: "TestSchema.Class" } as unknown as ECClass;
+        const property = { name: "PropertyName", isPrimitive: () => false } as unknown as ECProperty;
+        classStubs.stubEntityClass({
+          schemaName: "TestSchema",
+          className: "Class",
+          is: async () => true,
+          properties: [property],
+        });
+        const propertyInfo: propertiesGrouping.PropertyGroupInfo = {
+          ecClass,
+          previousPropertiesGroupingInfo: [],
+          propertyGroup: { propertyName: "PropertyName" },
+        };
+        expect(await propertiesGrouping.createPropertyGroups(metadataProvider, nodes, propertyInfo, formatter)).to.deep.eq({
+          groupingType: "property",
+          grouped: [],
+          ungrouped: nodes,
+        });
+      });
+
+      it("doesn't group, when property value isn't set and createGroupForUnspecifiedValues isn't set", async () => {
+        const nodes = [
+          createTestProcessedInstanceNode({
+            key: { type: "instances", instanceKeys: [{ className: "TestSchema.Class", id: "0x1" }] },
+            processingParams: {
+              grouping: {
+                byProperties: {
+                  fullClassName: "TestSchema.Class",
+                  propertyGroups: [{ propertyName: "PropertyName" }],
```

Comment:
maybe set `propertyValue: undefined` to clarify where the value comes from

Reply:
Added.